massgen/tests/integration/test_orchestrator_restart_and_external_tools.py

```
@@ -720,6 +721,82 @@ def test_setup_hook_manager_for_codex_hybrid_sets_native_and_mcp(mock_orchestrat
+    agent.backend.supports_native_hooks = lambda: True
+    agent.backend.supports_mcp_server_hooks = lambda: True
+    agent.backend.get_native_hook_adapter = lambda: adapter
+    agent.backend.set_native_hooks_config = lambda _config: None
+    agent.backend.set_background_wait_interrupt_provider = lambda _provider: None
+
+    written_payloads: list[str] = []
+    setattr(
+        agent.backend,
+        "write_post_tool_use_hook",
+        lambda content, tool_matcher="*", ttl_seconds=30.0: written_payloads.append(content),
+    )
```

Ruff (0.15.10)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)